### PR TITLE
Diverse

### LINKED ISSUE #230: Change from types in docstring to proper typehints
since this is way more helpful

### PR DESCRIPTION
**Pull request by yannikschaelte**
* add typehints to most global functions and classes
* tidy up some stuff
* rename GlobalOptimizer -> PyswarmOptimizer
* add rdatas_to_simulation_df function

fixes #230

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/options.py  2
- pypesto/optimize/result.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5221283&bid=17119763)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/options.py  2
- pypesto/optimize/result.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5221283&bid=17119763)

File: pypesto/engine/task.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def __init__(self):
         pass
 
     @abc.abstractmethod
-    def execute(self) -> 'OptimizerResult':  # pylint: disable=R0201
+    def execute(self) -> 'OptimizerResult':  # noqa: F821,R0201
```
Comment (FFroehlich) [inline pypesto/engine/task.py]: I believe a better way to address is to make sure forward imports are actually well defined. See https://www.python.org/dev/peps/pep-0484/#forward-references

Comment (yannikschaelte) [reply pypesto/engine/task.py]: agreed. is it now like what you meant?

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/result.py  4
- pypesto/optimize/options.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5221283&bid=17119763)

File: pypesto/engine/task.py (hunk 1 above)
Comment (FFroehlich) [reply pypesto/engine/task.py]: yes :)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/result.py  4
- pypesto/optimize/options.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5221283&bid=17119763)

**Pull request merged**